### ISSUE TITLE
If a category has exact name entered, show it first

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#844]
WIth the latest Git, yesterday I got this situation: I typed the name of a category that I know exists, but I had a hard time finding it, because it was far down in the suggestions.

My suggestion: If a category has _exact_ name entered (ignoring uppercase/lowercase), show it first

**Comment by misaochan** [comment]
Agreed, that would be good!

**Comment by ismailisimba** [comment]
You could design the search method to check for similarities ignoring caps, then by simple calculation decide if by how much percentage the query matches a category. Then sort the results from highest match to lowest.

**Comment by nicolas-raoul** [comment]
@ismailisimba Something similar has already been implemented by @misaochan, please see https://github.com/commons-app/apps-android-commons/wiki/Category-suggestions-%28readme%29
Cheers!

**Comment by Kartik2016** [comment]
Is the issue still open? if yes, can i work on it ?

**Comment by pszklarska** [comment]
If @Kartik2016 doesn't work on this issue, I'd like to fix this :)

**Comment by nicolas-raoul** [comment]
Oh, sorry for not assigning this one in time...
It seems that both @Kartik2016 and @pszklarska started working on it at the same time. Apologies for the confusion, this time I guess it makes sense to assign it to @pszklarska who has sent a pull request already. In the future we will try to assign sooner!

**Comment by nicolas-raoul** [comment]
Hum, looks like Github does not allow me to assign to anybody outside the organization, actually!
This means we need another way to indicate that an issue is assigned... maybe a label?
In any case we will need to make that explicit in [the documentation](https://github.com/commons-app/apps-android-commons/wiki/Volunteers-welcome%21).

**Comment by misaochan** [comment]
Yeah, the assignment is messy. :/ Maybe we could just state in comments "assigned to [name]", along with adding that label?

**Comment by nicolas-raoul** [comment]
Strangely I still see this problem recently, the category with the entered name is very far in the list of suggestions:
![screenshot_20180125-144605](https://user-images.githubusercontent.com/99590/36718635-e13713fc-1be5-11e8-925f-addbc962b181.png)

**Comment by gupta-meghna64** [comment]
@nicolas-raoul This is strangely weird. I tried with exactly the ones you tried, and they're appearing at the top of list for me :-/

**Comment by nicolas-raoul** [comment]
@gupta-meghna64 I am using the Play Store version, I guess this has been fixed in master? :-)

**Comment by gupta-meghna64** [comment]
Yes @nicolas-raoul :)

**Comment by nicolas-raoul** [comment]
Happening currently in 2.8-release branch:
![screenshot_20180720-004644](https://user-images.githubusercontent.com/99590/43031690-c5bda196-8ce1-11e8-9060-89260f02c93c.png)
